website/src/controllers/api-v3/user.js

```diff
@@ -31,4 +39,98 @@ api.getUser = {
   },
 };
 
+let partyMembersFields = 'profile.name stats achievements items.special';
+
+/**
+ * @api {post} /user/class/cast/:spell Cast a spell on a target.
+ * @apiVersion 3.0.0
+ * @apiName UserCast
+ * @apiGroup User
+ *
+ * @apiParam {string} spellId The spell to cast.
+ * @apiParam {string="party","self","user","task"} targetType Query parameter, the target type for the spell.
+ * @apiParam {UUID} targetId Optional query parameter, the id of the target when casting a spell on a party member or a task.
+ *
+ * @apiSuccess {Object|Array} mixed Will return the modified targets. For party members only the necessary fields will be populated.
+ */
+api.castSpell = {
+  method: 'POST',
+  middlewares: [authWithHeaders(), cron],
+  url: '/user/class/cast/:spell',
+  async handler (req, res) {
+    let user = res.locals.user;
+    let targetType = req.query.targetType;
```

Comment:
In the newest commit I removed targetType since we store it in the spell object. So at this point for targetId only it doesn't make sense for me to use req.body